pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -545,90 +553,161 @@ public boolean jdbcCompliant() {
   public static Properties parseURL(String url, Properties defaults) {
     Properties urlProps = new Properties(defaults);
 
-    String urlServer = url;
-    String urlArgs = "";
-
-    int qPos = url.indexOf('?');
-    if (qPos != -1) {
-      urlServer = url.substring(0, qPos);
-      urlArgs = url.substring(qPos + 1);
-    }
-
-    if (!urlServer.startsWith("jdbc:postgresql:")) {
+    if (!url.startsWith("jdbc:postgresql:")) {
```

Comment:
Prefer you did not modify the argument to the function

Reply:
> Prefer you did not modify the argument to the function

I did because url can now be an ldapUrl so I moved the parsing at the end

    int qPos = url.indexOf('?');
    if (qPos != -1) {
      urlServer = url.substring(0, qPos);
      urlArgs = url.substring(qPos + 1);
    }

Reply:
I had to move it after because the LDAP can also return a url... that contains urlArgs... see what I mean ?

Reply:
so how does https://github.com/pgjdbc/pgjdbc/blob/c61112b5f80072a53d422963b271079f48626f0a/pgjdbc/src/main/java/org/postgresql/Driver.java#L601 work then since you have removed this part of the url.